auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -186,36 +186,37 @@ private void populateIdpList(List<IdpConfig> providerConfigs) {
         for (IdpConfig idpConfig : providerConfigs) {
             @LayoutRes int buttonLayout;
 
-            if (TextUtils.isEmpty(idpConfig.getParams().getString(GENERIC_OAUTH_PROVIDER_ID))) {
-                final String providerId = idpConfig.getProviderId();
-                switch (providerId) {
-                    case GoogleAuthProvider.PROVIDER_ID:
-                        buttonLayout = R.layout.fui_idp_button_google;
-                        break;
-                    case FacebookAuthProvider.PROVIDER_ID:
-                        buttonLayout = R.layout.fui_idp_button_facebook;
-                        break;
-                    case TwitterAuthProvider.PROVIDER_ID:
-                        buttonLayout = R.layout.fui_idp_button_twitter;
-                        break;
-                    case GithubAuthProvider.PROVIDER_ID:
-                        buttonLayout = R.layout.fui_idp_button_github;
-                        break;
-                    case EMAIL_LINK_PROVIDER:
-                    case EmailAuthProvider.PROVIDER_ID:
-                        buttonLayout = R.layout.fui_provider_button_email;
-                        break;
-                    case PhoneAuthProvider.PROVIDER_ID:
-                        buttonLayout = R.layout.fui_provider_button_phone;
-                        break;
-                    case AuthUI.ANONYMOUS_PROVIDER:
-                        buttonLayout = R.layout.fui_provider_button_anonymous;
+            final String providerId = idpConfig.getProviderId();
+            switch (providerId) {
+                case GoogleAuthProvider.PROVIDER_ID:
+                    buttonLayout = R.layout.fui_idp_button_google;
+                    break;
+                case FacebookAuthProvider.PROVIDER_ID:
+                    buttonLayout = R.layout.fui_idp_button_facebook;
+                    break;
+                case TwitterAuthProvider.PROVIDER_ID:
+                    buttonLayout = R.layout.fui_idp_button_twitter;
+                    break;
+                case GithubAuthProvider.PROVIDER_ID:
+                    buttonLayout = R.layout.fui_idp_button_github;
+                    break;
+                case EMAIL_LINK_PROVIDER:
+                case EmailAuthProvider.PROVIDER_ID:
+                    buttonLayout = R.layout.fui_provider_button_email;
+                    break;
+                case PhoneAuthProvider.PROVIDER_ID:
+                    buttonLayout = R.layout.fui_provider_button_phone;
+                    break;
+                case AuthUI.ANONYMOUS_PROVIDER:
+                    buttonLayout = R.layout.fui_provider_button_anonymous;
+                    break;
+                default:
```

Comment:
Could this actually be:
```java
case AuthUI.APPLE_PROVIDER:
case AuthUI.MICROSOFT_PROVIDER:
case AuthUI.YAHOO_PROVIDER:
  buttonLayout = idpConfig.getParams().getInt(GENERIC_OAUTH_BUTTON_ID);
  break;
default:
  throw new IllegalStateException("Unknown provider: " + providerId);

Reply:
Hmm, the only benefit I see here is that it makes the code slightly cleaner. I don't think it's worth it. It'll be another thing to remove once we decide to support all idps.